decompressed/base/etc/init.d/rootdevice

```diff
@@ -165,24 +165,36 @@ check_and_apply_root() {
     store_pass_in_uci
   fi
 
-  #TODO: add more checks to be sure our dropbear config is not overrided from another
-  # use arbitrary @dropbear[-1]? fully clean dropbear file?
-  logger_command "Setting & Enabling Dropbear SSH Server"
-  dropbear_path="dropbear.@dropbear[0]"
-  if [ "$(uci get -q dropbear.lan)" ]; then
+  logger_command "Checking and force-enabling dropbear SSH Server for root on lan"
+  dropbear_path="dropbear.@dropbear[-1]"
+  if [ "$(uci get -q dropbear.afg)" ]; then
+    dropbear_path="dropbear.afg"
+  elif [ "$(uci get -q dropbear.lan)" ]; then
     dropbear_path="dropbear.lan"
   fi
-  uci set "$dropbear_path".Interface='lan'
-  uci set "$dropbear_path".RootLogin='1'
-  uci set "$dropbear_path".RootPasswordAuth='on' #dropbear root related
-  uci set "$dropbear_path".PasswordAuth='on'
-  uci set "$dropbear_path".enable='1'
-
-  uci commit dropbear
-  uci commit modgui
+  if [ ! "$(uci get -q dropbear.@dropbear[-1])" ]; then #in case of dropbear config empty we must add a section
```

Comment:
Qua c'è qualcosa che non va. Questo dovrebbe essere l'else dell'if precedente, rinominare la sezione appena creata con un alias e impostare dropbear_path a dropbear._alias_

Altrimenti qua possono succedere cose strane, tipo lui potrebbe trovare solo una sezione dropbear.wan e usare quella per impostare l'accesso lan.

Reply:
diciamo che era quello che volevo fare modificare una sezione esistente se c'era, ma a questo punto se quella che c'è è una wan meglio lasciarla intatta, non credo ci sia bisogno poi di modificare dropbear_path se messo in un else, con [-1] dovrebbe comunque andare a modificare quella che ha appena aggiunto

Reply:
No tu in generale non dovresti mai riutilizzare quelle che ci sono perchè quelle che ci sono vengono create in automatico dal firmware dai valori di default o anche dalla telegestione perchè quelli sono percorsi di configurazione noti che l'isp sa di poter usare e si avere a disposizione sui suoi firmware. Semplificherei il tutto con un bel "se esiste una sezione afg usa quella altrimenti creala e usala. In afg e tch-exploit al posto di controllare se esiste viene prima rimossa e poi ricreata, così facendo garantiscono che non esistano altri parametri aggiuntivi in quella stessa sezione. Quello che hai fatto qui, cioè riutilizzare una sezione esistente senza prima resettarla potrebbe funzionare in modo ugualmente affidabile solo se reimposti tutti i possibili parametri di quella sezione, e questo è difficile perchè i possibili parametri potrebbero cambiare da un momento all'altro, e questo è vero per tutti i file di uci, non solo per dropbear che forse è quello più "stabile"..

Reply:
quello di reimpostre è una cosa che già faccio nel codice attuale, e inoltre viene assicurata sia consistente ad ogni avvio (questo codice è eseguito ad ogni riavvio) riutilizzare almeno il blocco lan vorrei lasciarlo per il semplice motivo che altrimenti si creerebbe un'inconsistenza di configurazione con il blocco afg e il blocco lan che ascoltano sulla stessa interfaccia sulla stessa porta

Reply:
Si ma tu non le reimposti tutte, dovresti farlo per garantire quello che dicevo prima. E' chiaro che al momento non sembrano esserci altre opzioni definite per dropbear che possano creare casino se differentemente impostate.

La storia dei conflitti potresti risolverla così:
```bash
disable_existing_dropbear() {
        [ "${enable}" = "1" ] && uci set dropbear.${1}.enable=0
}
        ...
        . /lib/functions.sh
        config_load "dropbear"
        config_foreach disable_existing_dropbear dropbear
```

Reply:
Bhè io setto tutte quelle che potrebbero impedire un accesso root.. o no?

in quel modo disabilito forzatamente (già ora significa che sulla lan si mette forzatamente in ascolto sulla 22 che comunque non è una cosa bella) anche gli accessi ssh dell'ISP e/o di altro inserito dall'utente che non è mai stata una cosa di cui la GUI si occupa (anzi fa esattamente il contrario, cercando di simulare sia tutto OK alla telegestione)

Reply:
Se l'ISP crea solo una singola sezione di dropbear, non vi imposta nessuna interface (bind su 0.0.0.0), lascia la porta di default (ovvero 22) e blocca da firewall l'accesso locale a quella porta, sei già costretto ad intervenire su di essa perchè non riusciresti a mettere su una seconda istanza tipo 192.168.1.1:22 quando ne esiste già una su 0.0.0.0:22.

Se vuoi garantire il non conflitto (e quindi garantire accesso all'utente) devi per forza di cose disabilitare tutte quelle che hanno:
- Port vuota (quindi 22) o "22" e
  - Interface vuota (non impostata, quindi bind su 0.0.0.0) o Interface impostata su "lan"

Questo però non ti garantisce di non andare a disabilitare un'istanza potenzialmente usata dall'ISP configurata come detto sopra. Viceversa se disabiliti solo quelle con Interface "lan" forse preservi quelle utili all'ISP ma non garantisci che la nuova creata/riciclata non vada in conflitto con l'altra. Tra garantire l'accesso all'utente e garantire quello all'ISP credo sia qui opportuno optare per la prima soluzione

```bash
disable_existing_dropbear() {
        [ "${enable}" = "1" ] && ([ "${Interface}" = "lan" ] || [ "${Interface}" = "" ]) && ([ "${Port}" = "22" ] || [ "${Port}" = "" ]) && uci set dropbear.${1}.enable=0
}
```

Reply:
Anzi meglio così, perchè di default enable vuota equivale a 1

```bash
disable_existing_dropbear() {
        [ "${enable}" != "0" ] \
        && ([ "${Interface}" = "lan" ] || [ "${Interface}" = "" ]) \
        && ([ "${Port}" = "22" ] || [ "${Port}" = "" ]) \
        && uci set dropbear.${1}.enable=0
}
```

Reply:
ok vedi se ora il ragionamento fila..